server/src/main/java/org/opensearch/common/blobstore/transfer/RemoteTransferContainer.java

```diff
@@ -0,0 +1,252 @@
+     * @param remoteFileName                 Name of the remote file
+     * @param contentLength                  Total content length of the file to be uploaded
+     * @param failTransferIfFileExists       A boolean to determine if upload has to be failed if file exists
+     * @param writePriority                  The {@link WritePriority} of current upload
+     * @param offsetRangeInputStreamSupplier A supplier to create OffsetRangeInputStreams
+     * @param isRemoteDataIntegritySupported A boolean to signify whether the remote repository supports server side data integrity verification
+     * @param areInputStreamsDecorated            A boolean to signify whether the streams created via {@link OffsetRangeInputStreamSupplier#get} are decorated or not
+     */
+    public RemoteTransferContainer(
+        String fileName,
+        String remoteFileName,
+        long contentLength,
+        boolean failTransferIfFileExists,
+        WritePriority writePriority,
+        OffsetRangeInputStreamSupplier offsetRangeInputStreamSupplier,
+        long expectedChecksum,
+        boolean isRemoteDataIntegritySupported,
+        boolean areInputStreamsDecorated
+    ) {
+        this.fileName = fileName;
+        this.remoteFileName = remoteFileName;
+        this.contentLength = contentLength;
+        this.failTransferIfFileExists = failTransferIfFileExists;
+        this.writePriority = writePriority;
+        this.offsetRangeInputStreamSupplier = offsetRangeInputStreamSupplier;
+        this.expectedChecksum = expectedChecksum;
+        this.isRemoteDataIntegritySupported = isRemoteDataIntegritySupported;
+        this.areInputStreamsDecorated = areInputStreamsDecorated;
+    }
+
+    /**
+     * @return The {@link  WriteContext} for the current upload
+     */
+    public WriteContext createWriteContext() {
+        return new WriteContext(
+            remoteFileName,
+            this::supplyStreamContext,
+            contentLength,
+            failTransferIfFileExists,
+            writePriority,
+            this::finalizeUpload,
+            isRemoteDataIntegrityCheckPossible(),
+            isRemoteDataIntegrityCheckPossible() ? expectedChecksum : null
+        );
+    }
+
+    /**
+     * This method is called to create the {@link StreamContext} object that will be used by the vendor plugin to
+     * open streams during uploads. Calling this method won't actually create the streams, for that the consumer needs
+     * to call {@link StreamContext#provideStream}
+     *
+     * @param partSize Part sizes of all parts apart from the last one, which is determined internally
+     * @return The {@link StreamContext} object that will be used by the vendor plugin to retrieve streams during upload
+     */
+    public StreamContext supplyStreamContext(long partSize) {
+        try {
+            return this.openMultipartStreams(partSize);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
Can this be private instead?

Reply:
I think yes. @raghuvanshraj

Reply:
I think yes. @raghuvanshraj

Reply:
I've made it package private as it is being used in tests.